Sources/TuistGenerator/Generator/LinkGenerator.swift

```diff
@@ -224,21 +224,29 @@ final class LinkGenerator: LinkGenerating {
         pbxTarget.buildPhases.append(buildPhase)
 
         try dependencies.forEach { dependency in
-            if case let DependencyReference.absolute(path) = dependency {
+            switch dependency {
+            case let .absolute(path):
                 guard let fileRef = fileElements.file(path: path) else {
                     throw LinkGeneratorError.missingReference(path: path)
                 }
                 let buildFile = PBXBuildFile(file: fileRef)
                 pbxproj.add(object: buildFile)
                 buildPhase.files?.append(buildFile)
-
-            } else if case let DependencyReference.product(name) = dependency {
+            case let .product(name):
                 guard let fileRef = fileElements.product(name: name) else {
                     throw LinkGeneratorError.missingProduct(name: name)
                 }
                 let buildFile = PBXBuildFile(file: fileRef)
                 pbxproj.add(object: buildFile)
                 buildPhase.files?.append(buildFile)
+            case let .sdk(sdkPath, sdkStatus):
+                guard let fileRef = fileElements.sdk(path: sdkPath) else {
+                    throw LinkGeneratorError.missingReference(path: sdkPath)
```

Comment:
Since there's logic responsible for adding all the file elements to the project, I think it's safe not throwing here and assume that the file is there. By throwing we present the developers with a non-actionable error that represents a bug in our side, and we should avoid that.

Reply:
You are right, these are non-actionable errors :( 

We've had a few cases in the past where this would occur due to a few bugs (e.g. https://github.com/tuist/tuist/issues/352, https://github.com/tuist/tuist/issues/345, https://github.com/tuist/tuist/issues/322, https://github.com/tuist/tuist/issues/330) - perhaps crashing may not provide the best experience either. 

How about we change the error type to `.bug`? It's been super helpful in those issues when users posted that particular error message - it helped us pinpoint the cause and fix those issues :)

Reply:
it's already a `.bug` :)

Reply:
Fair point then. Although we are not doing anything with those bugs. I [opened an issue](https://github.com/tuist/tuist/issues/413) to add error reporting.

Reply:
👍